lib/elephant_in_the_room/sites/site.ex

```diff
@@ -18,9 +24,62 @@ defmodule ElephantInTheRoom.Sites.Site do
   @doc false
   def changeset(%Site{} = site, attrs) do
     site
-    |> cast(attrs, [:name, :host])
+    |> cast(attrs, [:name, :host, :description, :title])
     |> validate_required([:name, :host])
     |> unique_constraint(:name)
     |> unique_constraint(:host)
+    |> store_image(attrs)
+    |> validate_favicon(attrs)
+    |> check_title(attrs)
+  end
+
+  def check_title(%Changeset{} = changeset, %{"title" => _title}), do: changeset
+
+  def check_title(%Changeset{} = changeset, %{"name" => name}),
+    do: Map.put(changeset, :title, name)
+
+  def check_title(%Changeset{} = changeset, _attrs), do: changeset
+
+  def store_image(%Changeset{} = changeset, %{"image" => nil}),
+    do: put_change(changeset, :image, nil)
+
+  def store_image(%Changeset{} = changeset, %{"image" => image}) do
+    {:ok, image_name} = Image.store(%{image | filename: Ecto.UUID.generate()})
+
+    put_change(changeset, :image, "/images/#{image_name}")
+  end
+
+  def store_image(%Changeset{} = changeset, _attrs), do: changeset
+
+  def validate_favicon(%Changeset{} = changeset, %{"favicon" => nil}),
+    do: put_change(changeset, :image, nil)
+
+  def validate_favicon(
+        %Changeset{} = changeset,
+        %{"favicon" => %Plug.Upload{content_type: ct} = favicon}
+      ) do
+    if ct == "image/x-icon" do
```

Comment:
I'm getting the `image/vnd.microsoft.icon` content type. Therefore I'm unable to upload icons.